src/protocol/sort/reshare.rs

```diff
@@ -0,0 +1,138 @@
+use crate::{
+    error::BoxError,
+    field::Field,
+    helpers::{
+        mesh::{Gateway, Mesh},
+        prss::SpaceIndex,
+        Direction, Identity,
+    },
+    protocol::{context::ProtocolContext, RecordId, Step},
+    secret_sharing::Replicated,
+};
+use embed_doc_image::embed_doc_image;
+use serde::{Deserialize, Serialize};
+
+/// A message sent by each helper when they've reshared their own shares
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct PartValue<F> {
+    part: F,
+}
+/// Reshare(i, [x])
+// This implements reshare algorithm of "Efficient Secure Three-Party Sorting Protocol with an Honest Majority" at communication cost of 2R.
+// Input: Pi-1 and Pi+1 knows their secret shares
+// Output: At the end of the protocol, [all 3 helpers receive renewed shares of x with randomness added
```

Comment:
spurious "[", please remove

Reply:
Also, I'm not sure what you mean by "with randomness added". That sounds like we added a random number to the secret-shared value, which isn't what's going on. I'd say: "...all 3 helpers receive their shares of a new, random secret sharing of the secret value".